### PR TITLE
Page order rebuild

### PR DESCRIPTION
**Pull request by markhannum**
First part of page-order rebuild code.  This is useful for the extremely rare cases where a btree is corrupt and has a cycle in it's leaf pages.  Rebuilding in page order mode requires that the master drop into READONLY mode.

### CODE REVIEW

**Comment by markhannum** [comment]
Yes- completely forgot.  Thanks Rivers!

**Comment by riverszhang89** [comment]
Whoah roborivers so smart!

**Comment by markhannum** [comment]
The format for rebuild.gif looks different than the original..

File: schemachange/sc_alter_table.c [hunk 1]
```diff
@@ -542,8 +544,13 @@ int do_alter_table(struct ireq *iq, tran_type *tran)
      * that doesn't require rebuilding anything. */
     if ((!newdb->plan || newdb->plan->plan_convert) ||
         changed == SC_CONSTRAINT_CHANGE) {
+        readonly = !s->live;
         doing_conversion = 1;
+        if (readonly)
+            gbl_readonly_sc = 1;
         rc = convert_all_records(db, newdb, newdb->sc_genids, s);
+        if (readonly)
+            gbl_readonly_sc = 0;
```
Comment (ldennis) [inline schemachange/sc_alter_table.c]: defer resetting gbl_readonly_sc after the sc commits

File: schemachange/sc_alter_table.c [hunk 2]
```diff
@@ -542,8 +544,13 @@ int do_alter_table(struct ireq *iq, tran_type *tran)
      * that doesn't require rebuilding anything. */
     if ((!newdb->plan || newdb->plan->plan_convert) ||
         changed == SC_CONSTRAINT_CHANGE) {
+        readonly = !s->live;
```
Comment (ldennis) [inline schemachange/sc_alter_table.c]: use `s->live`

**Comment by riverszhang89** [comment]
> Custom tokens and rules must be defined after SQLite's.
> Please update SQL documentation and bubble diagrams to reflect the changes you've made, if necessary.

Please ignore those - Roborivers blindly looks at the rules defined in https://github.com/riverszhang89/review-helper and post them.

**Pull request merged**